tutorials/intermediate/imshow_extent.py

```diff
@@ -0,0 +1,179 @@
+            bbox_kwargs = {'fc': 'w', 'alpha': .75, 'boxstyle': "round4"}
+            ann_kwargs = {'xycoords': 'axes fraction',
+                          'textcoords': 'offset points',
+                          'bbox': bbox_kwargs}
+
+            ax.annotate(upper_string, xy=(.5, 1), xytext=(0, -1),
+                        ha='center', va='top', **ann_kwargs)
+            ax.annotate(lower_string, xy=(.5, 0), xytext=(0, 1),
+                        ha='center', va='bottom', **ann_kwargs)
+
+            ax.annotate(port_string, xy=(0, .5), xytext=(1, 0),
+                        ha='left', va='center', rotation=90,
+                        **ann_kwargs)
+            ax.annotate(starboard_string, xy=(1, .5), xytext=(-1, 0),
+                        ha='right', va='center', rotation=-90,
+                        **ann_kwargs)
+
+            ax.set_title('origin: {origin}'.format(origin=origin))
+
+            if not auto_limits:
+                ax.set_xlim(-1, 7)
+                ax.set_ylim(-1, 6)
+
+    for ax, extent in zip(columns['label'], extents):
+        text_kwargs = {'ha': 'right',
+                       'va': 'center',
+                       'xycoords': 'axes fraction',
+                       'xy': (1, .5)}
+        if extent is None:
+            ax.annotate('None', **text_kwargs)
+            ax.set_title('`extent=`')
+        else:
+            left, right, bottom, top = extent
+            text = ('left: {left:0.1f}\nright: {right:0.1f}\n' +
+                    'bottom: {bottom:0.1f}\ntop: {top:0.1f}\n').format(
+                        left=left, right=right, bottom=bottom, top=top)
+
+            ax.annotate(text, **text_kwargs)
+        ax.axis('off')
+
+
+extents = (None,
+           (-0.5, 6.5, -0.5, 5.5),
+           (-0.5, 6.5, 5.5, -0.5),
+           (6.5, -0.5, -0.5, 5.5),
+           (6.5, -0.5, 5.5, -0.5))
+
+###############################################################################
+#
+#
+# First, using *extent* we pick a bounding box in dataspace that the
+# image will fill and then interpolate/resample the underlying data to
+# fill that box.
+#
+# - If ``origin='lower'`` than the ``[0, 0]`` entry is closest to the
+#   ``(left, bottom)`` corner of the bounding box and moving closer to
+#   ``(left, top)`` moves along the ``[:, 0]`` axis of the array to
+#   higher indexed rows and moving towards ``(right, bottom)`` moves you
+#   along the ``[0, :]`` axis of the array to higher indexed columns
+#
+# - If ``origin='upper'`` then the ``[-1, 0]`` entry is closest to the
+#   ``(left, bottom)`` corner of the bounding box and moving towards
+#   ``(left, top)`` moves along the ``[:, 0]`` axis of the array to
+#   lower index rows and moving towards ``(right, bottom)`` moves you
+#   along the ``[-1, :]`` axis of the array to higher indexed columns
+#
+# To demonstrate this we will plot a linear ramp
+# ``np.arange(42).reshape(6, 7)`` with varying parameters.
+#
+
+generate_imshow_demo_grid(True, extents[:1])
```

Comment:
There isn't a massive change in these images from L --> R, but a clear different from top to bottom. Maybe a good place to use a colourmap that isn't perceptually uniform to illustrate changes in orientation?

Reply:
At least when I look a these what I see is the direction of highest-lowest vector, changed to a diagonal gradient.